internal/environment/environment.go

```diff
@@ -173,6 +169,11 @@ func (n *NomadEnvironment) SetNetworkAccess(allow bool, exposedPorts []uint16) {
 // Register creates a Nomad job based on the default job configuration and the given parameters.
 // It registers the job with Nomad and waits until the registration completes.
 func (n *NomadEnvironment) Register(apiClient nomad.ExecutorAPI) error {
+	// To avoid docker image issues. See https://github.com/openHPI/poseidon/issues/69
+	if err := n.Delete(apiClient); err != nil {
+		return fmt.Errorf("failed to remove the environment: %w", err)
+	}
+
```

Comment:
I'm not sure whether this is the correct place. I would argue that removing the environment in a method called "Register" is unexpected. Wouldn't it make sense to put these lines to the manager? 

https://github.com/openHPI/poseidon/blob/dd7b150e55b9b91c6d0b3edae0c5e50fa3f91de1/internal/environment/manager.go#L133

Maybe we could also split the `CreateOrUpdate` method to make the purpose even more clear:

```go
// Fetch or create execution environment
environment, existing_environment := m.runnerManager.GetEnvironment(id)
if existing_environment {
    // Remove existing environment to force downloading the newest Docker image
    // See https://github.com/openHPI/poseidon/issues/69
    err = environment.Delete(m.api)
    if err != nil {
        return fmt.Errorf("failed to remove the environment: %w", err)
     }
}

// Create a new environment with the given ID // could be a dedicated method?
environment, err = NewNomadEnvironment(m.templateEnvironmentHCL)
if err != nil {
    return false, fmt.Errorf("error creating Nomad environment: %w", err)
}
environment.SetID(id)


// Set options according to request // maybe extract to separate method or together with `NewNomadEnvironment`  part?
environment.SetPrewarmingPoolSize(request.PrewarmingPoolSize)
environment.SetCPULimit(request.CPULimit)
environment.SetMemoryLimit(request.MemoryLimit)
environment.SetImage(request.Image)
environment.SetNetworkAccess(request.NetworkAccess, request.ExposedPorts)

// Keep a copy of environment specification in memory // maybe rename method and remove return value (isn't this similar to `!existing_environment `)?
m.runnerManager.StoreEnvironment(environment)

// Register template Job with Nomad
err = environment.Register(m.api)
if err != nil {
    return false, fmt.Errorf("error registering template job in API: %w", err)
}

// Launch idle runners based on template job
err = environment.LaunchIdleRunners(m.api) // maybe rename method?
if err != nil {
    return false, fmt.Errorf("error scaling template job in API: %w", err)
}

return existing_environment, nil
```

Just a proposal to have the important things in one place for creating / updating runners. I didn't test the code, so take it with a grain of salt.